src/libstd/sys/unix/mod.rs

```diff
@@ -58,6 +59,20 @@ pub fn init() {
     unsafe {
         assert!(signal(libc::SIGPIPE, libc::SIG_IGN) != !0);
     }
+
+    // A nicer handler for out-of-memory situations than the default one. This
+    // one prints a message to stderr before aborting.
```

Comment:
Could you expand this to explain:

* It's critical that this function doesn't have any allocations, so it shouldn't be using many abstractions.
* Errors are ignored here because there's not really anything else that can be done.